website/blog/2019-07-15-complete-table-scan.md

```diff
@@ -0,0 +1,169 @@
+---
+title: Complete Table Scan: A Quantitative Assessment
+author: Orri Erling
+authorURL: http://code.fb.com/
+authorFBID: 100026224749124
+---
+
+In the previous article we looked at the abstract problem statement and possibilities inherent in scanning tables. In this piece we look at the quantitative upside with Presto. We look at a number of queries and explain the findings.
+
+The initial impulse motivating this work is the observation that table scan is by far the #1 operator in Presto workloads I have seen. This is a little over half of all Presto CPU, with repartitioning a distant second, at around 1/10 of the total. The other half of the motivation is ready opportunity: Presto in its pre-Aria state does almost none of the things that are common in table scan.
+
+For easy reproducibility and staying away from proprietary material, we use a TPC-H 100G dataset running on a desktop machine with 2x4 Skylake cores at 3.5GHz. The data is compressed with Snappy and we are running with warm OS cache. The Presto is a modified 0.221 where the Aria functionality can be switched on and off. Not to worry, we will talk about disaggregated storage and IO in due time but the basics will come first.
+
+# Simple scan
+
+The base case for scan optimization is the simplest possible query:
+
+```sql
+SELECT SUM(l_extendedprice)
+FROM lineitem
+WHERE suppkey = 12345;
+```
+| Version  | Wall time (seconds) | CPU time (seconds) |
```

Comment:
@aweisberg 

I'd pivot this table and a column for ration between aria and baseline. I think this will be clearer.

Reply:
Added Baseline CPU / Aria CPU column.